pkg/models/tagged_html.go

```diff
@@ -0,0 +1,207 @@
+package models
+
+import (
+	"context"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"regexp"
+
+	"github.com/samber/lo"
+
+	"github.com/cms-enterprise/easi-app/pkg/appcontext"
+	"github.com/cms-enterprise/easi-app/pkg/sanitization"
+)
+
+var (
+	spanRe      = regexp.MustCompile(`<span[^>]*>.*?</span>`)
+	attributeRe = regexp.MustCompile(`(\S+)="([^"]+)"`)
+)
+
+// TaggedHTML Is the input type for HTML that could contain tags
+type TaggedHTML TaggedContent
+
+// TaggedContent represents rich text HTML with possible tagged HTML
+type TaggedContent struct {
+	RawContent HTML
+	Tags       []*Tag
```

Comment:
I noticed that this deviates from MINT a bit in the naming and use of Tags vs the intermediary Mentions Structure. Not necessarily an issue, but would you mind clarifying your reasons for the change? 

It would be good if a `Tags` doesn't mean something different in EASI than it does in MINT.

Reply:
Great question! Thanks Steven. Two reasons, the first of which is that we are not saving any tags/mentions to the DB, so we didn't need the second type, which looked like it was meant to represent a DB record. (BTW - we were deciding which naming convention to use, and we opted for `Tag` instead of `Mention` because that's what it has been referred to for a while)

Second reason, maybe you can help me on: I couldn't figure out a concrete difference between `Tag` and `Mention`, outside of one representing a parsed tag and the other representing a DB row. This appears to work with just one of the types, so we figured to not include the other one also